### ISSUE TITLE
All the small things

### ISSUE THREAD

**Issue by JosephTLyons** [zed-industries/feedback#52]
Here's a list of "smaller" things that I think will add a ton of value to the Zed experience:

- https://github.com/zed-industries/feedback/issues/48
- https://github.com/zed-industries/feedback/issues/50
- https://github.com/zed-industries/feedback/issues/49
- https://github.com/zed-industries/feedback/issues/48

**Comment by JosephTLyons** [comment]
Test 

- https://github.com/zed-industries/feedback/issues/65

**Comment by exalted** [comment]
:two-cents: You'll know better, but [a GitHub project](https://github.com/zed-industries/feedback/projects) sounds like a better solution.

**Comment by JosephTLyons** [comment]
> :two-cents: You'll know better, but [a GitHub project](https://github.com/zed-industries/feedback/projects) sounds like a better solution.

It would be nice to have something that is more directly supported by GitHub and less custom code involved, but I don't know if GitHub projects support the same fields.  I checked it out and wasn't able to find a way to sort on reactions and comments and such.  You can create custom fields in the project table, but you'd have to manually update those, or write another script to tap into them.  I'm not sure what the best solution is for generating a list like this - this is just my first shot.

**Comment by maxbrunsfeld** [comment]
This is very cool @JosephTLyons; thanks for setting it up.

**Comment by JosephTLyons** [comment]
> This is very cool @JosephTLyons; thanks for setting it up.

Yeah no problem

I'm currently running it manually; I don't have any experience with GitHub actions and I don't know want to assume there are enough Action Minutes to go around, but if I spent some time to learn it and if there were enough minutes to spare, I could try to automate it to run once a day.

I've done my best to optimize the script, it still takes about 5 seconds to run it because I can't seem to find a way to limit the query size using the GitHub Search syntax, so even though I only display 5 items per label, I have to pull all open issues per label and slice the list down to size locally.

**Comment by SeparateRecords** [comment]
> I have to pull all open issues per label and slice the list down to size locally

@JosephTLyons I think you should be able to use `?per_page=5` in the query, eg. `https://api.github.com/search/issues?per_page=5&q=...` -- at Fig we've had to fight the API to get it to [show _more_ than 100 issues](https://docs.github.com/en/rest/guides/traversing-with-pagination#changing-the-number-of-items-received) 😄

Also, should be possible to do this concurrently - if the script is public, I'd be happy to optimize it :)

It can _probably_ just run in a repo on your user account, using a [gh actions schedule](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#schedule). If the repo is public, you don't need to worry about it using up your minutes!

**Comment by JosephTLyons** [comment]
> > I have to pull all open issues per label and slice the list down to size locally
> 
> @JosephTLyons I think you should be able to use `?per_page=5` in the query, eg. `https://api.github.com/search/issues?per_page=5&q=...` -- I've had to fight the API to get [_more_ than 100 issues](https://docs.github.com/en/rest/guides/traversing-with-pagination#changing-the-number-of-items-received) for the fig support dashboard 😄
> 
> Also, should be possible to do this concurrently - if the script is public, I'd be happy to optimize it :)
> 
> It can _probably_ just run in a repo on your user account, using a [gh actions schedule](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#schedule). If the repo is public, you don't need to worry about it using up your minutes!

I'm currently using [PyGitHub](https://github.com/PyGithub/PyGithub) Github.  There are two general ways I found you can grab issues using it.  I haven't found how to limit the pages using it, as the search it uses is more like the search you have on the frontend on the issues page.  I probably missed something, but I felt like I couldn't get the "per_page" filter to work there:

https://github.com/zed-industries/feedback/blob/347ecc15c8b0f4f696d91075f9d9f2f558cad5d0/scripts/update_top_ranking_issues/main.py#L41

**Comment by JosephTLyons** [comment]
Need some opinions here.  Previously, I was just skipping all issues with these tags ("discussed", "meta", "needs info", "triage"}, as these type of labels are just optional ones; all issues with these labels have other "core" labels that are required.  
We are adding more more labels now and these seems these newer labels can either be optional or core labels, which means issues can show up in multiple places on this script.  Should I simply filter out duplicates that occur further down in the file, or should I allow duplicate files to show up in multiple label groups?

**Comment by iamnbutler** [comment]
Honestly, tags will probably be fast and loose for a bit while we figure out how to parse feedback. 

In the coming few weeks the fire hose will probably open up a lot, as I think we'll be increasing the amount of people with access by triple digits.

Suggestions are welcome for a framework to organize/parse feedback.

**Comment by JosephTLyons** [comment]
> Honestly, tags will probably be fast and loose for a bit while we figure out how to parse feedback. 
> 
> 
> 
> In the coming few weeks the fire hose will probably open up a lot, as I think we'll be increasing the amount of people with access by triple digits.
> 
> 
> 
> Suggestions are welcome for a framework to organize/parse feedback.

Yeah, this was just something I tossed together randomly without much thought; if it ends up not working with however the repository is organized, we can toss it.

**Comment by JosephTLyons** [comment]
This issue *should* be auto-generating now, every 12 hours UTC (the cron scheduler can't be set to any timezone specifically).  I also set it to rebuild after issues close.  Feel free to change the triggers, cron schedule, or timezones in the script to your liking.

**Comment by JosephTLyons** [comment]
I'm thinking it might make sense to remove using comments as part of the sorting. With likes, users can only add one, and no one can change the creation datetime, but anyone can comment multiple times to boost the issue up (to the top of the group of issues with the same number of likes).  I'm thinking just likes and creation datetime is enough to stabilize the sort and prevent others from artificially boosting an issue's importance.

**Comment by jamesmunns** [comment]
Hey there! Apologies if this is off-topic, is there somewhere I can :+1: linux support as a blocking issue? I do have a Mac, but don't use it as a primary development machine, which means that I probably won't have a chance to test out Zed until there is some kind of Linux client.

I can open an issue, but it seems explicitly documented that is a WIP feature :)

**Comment by JosephTLyons** [comment]
> Hey there! Apologies if this is off-topic, is there somewhere I can 👍 linux support as a blocking issue? I do have a Mac, but don't use it as a primary development machine, which means that I probably won't have a chance to test out Zed until there is some kind of Linux client.
> 
> I can open an issue, but it seems explicitly documented that is a WIP feature :)

Sure, I made issues here:

https://github.com/zed-industries/feedback/issues/171
https://github.com/zed-industries/feedback/issues/172

Feel free to put your 👍 there
